### ISSUE TITLE
Allow users to enter their own regular expressions

### ISSUE THREAD

**Issue by grenzor** [mrdokenny/History-AutoDelete#27]
Currently it looks like all user entered expressions are converted to regular expressions but this doesn't allow users to enter their own regular expressions. Doing so will be extremely useful for user-defined specificity for some rules.

**Comment by anewuser** [comment]
In the meantime, try [Historia](https://addons.mozilla.org/addon/historia/).

**Comment by LupinIII** [comment]
Yes, this should really be a feature. At least let me import a file with custom regular expressions in it and leave them alone. Currently they get rewritten as soon as they are imported. It is impossible to match a "?" currently.

**Comment by anewuser** [comment]
The last commit was over a year ago. You should try the extension I"ve suggested.

**Comment by LupinIII** [comment]
I tried it. That extension uses some kind of, but not "regular" JS regular expressions, so it's not what I'd like either. I also like that this extension tells you what expressions match a site, which the other one doesn't (at least I didn't see it doing that).

**Comment by anewuser** [comment]
Refer to the help section:

    Regular Expression capability to Filter Patterns was added in v1.2. Patterns must start and end with forward slash e.g. /abc (\d+)\.\d*/

**Comment by grahamperrin** [comment]
https://github.com/History-AutoDelete/History-AutoDelete/issues/35#issuecomment-363257563 suggests Historia.
